register/murmur3/multihash_murmur3.go

```diff
@@ -0,0 +1,40 @@
+/*
+	This package has no purpose except to perform registration of multihashes.
+
+	It is meant to be used as a side-effecting import, e.g.
+
+		import (
+			_ "github.com/multiformats/go-multihash/register/murmur3"
+		)
+
+	This package registers multihashes for murmur3
+*/
+package murmur3
+
+import (
+	"hash"
+
+	multihash "github.com/multiformats/go-multihash/core"
+	"github.com/spaolacci/murmur3"
+)
+
+func init() {
+	multihash.Register(multihash.MURMUR3X64_64, func() hash.Hash { return murmur64{murmur3.New64()} })
+}
+
+// A wrapper is needed to export the correct size, because murmur3 incorrectly advertises Hash64 as a 128bit hash.
+type murmur64 struct {
```

Comment:
nit: maybe this type name should be as verbose as the constants became, for consistency?